test/Npgsql.Tests/CopyTests.cs

```diff
@@ -15,7 +16,7 @@ public class CopyTests : TestBase
         #region issue 2257
 
         [Test, Description("Reproduce #2257")]
-        public void Issue2257()
+        public async Task Issue2257([Values(true, false)] bool isAsync)
```

Comment:
In general, rather than parameterizing each and every test, you can parameterize the entire test suite. See ReaderTests for an example.

However, we're not going to expose a public API accepting a bool async flag. We could still have InternalsVisibleTo for this test, although since the implementation for sync and async is identical - except for the actual I/O calls at the bottom of the stack - this is probably not worth it.